azurelinuxagent/common/osutil/default.py

```diff
@@ -682,23 +682,13 @@ def get_first_if(self):
 
         # for 64bit the size is 40 bytes
         # for 32bit the size is 32 bytes
-        python_arc = platform.architecture()[0]
-        struct_size = 32 if python_arc == '32bit' else 40
+        python_arch = platform.architecture()[0]
+        struct_size = 32 if python_arch == '32bit' else 40
 
-        sock = socket.socket(socket.AF_INET,
```

Comment:
It was easier to move this block of code into it's own method, and mock out that method than it was to mock out every low level call in this method.